src/Form/index.tsx

```diff
@@ -433,6 +434,23 @@ function Form({
     }
   }, [stepKey]);
 
+  useEffect(() => {
```

Comment:
this useEffect coudl maybe be moved into the offlineRequestHandler file as well, and a hook can be returned from the file. e.g. `useOfflineRequestHandler()`